gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/util/DefaultGraphManager.java

```diff
@@ -0,0 +1,216 @@
+        traversalSources.put(tsName, ts);
+    }
+
+    /**
+     * Get the {@link Graph} and {@link TraversalSource} list as a set of bindings.
+     */
+    public final Bindings getAsBindings() {
+        final Bindings bindings = new SimpleBindings();
+        graphs.forEach(bindings::put);
+        traversalSources.forEach(bindings::put);
+        return bindings;
+    }
+
+    /**
+     * Rollback transactions across all {@link Graph} objects.
+     */
+    public final void rollbackAll() {
+        graphs.entrySet().forEach(e -> {
+            final Graph graph = e.getValue();
+            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
+                graph.tx().rollback();
+        });
+    }
+
+    /**
+     * Selectively rollback transactions on the specified graphs or the graphs of traversal sources.
+     */
+    public final void rollback(final Set<String> graphSourceNamesToCloseTxOn) {
+        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.ROLLBACK);
+    }
+
+    /**
+     * Commit transactions across all {@link Graph} objects.
+     */
+    public final void commitAll() {
+        graphs.entrySet().forEach(e -> {
+            final Graph graph = e.getValue();
+            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
+                graph.tx().commit();
+        });
+    }
+
+    /**
+     * Selectively commit transactions on the specified graphs or the graphs of traversal sources.
+     */
+    public final void commit(final Set<String> graphSourceNamesToCloseTxOn) {
+        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.COMMIT);
+    }
+
+    /**
+     * If {@link Map} containing {@link Graph} references contains one corresponding to
+     * {@link String} graphName, then we return that {@link Graph}; otherwise, instantiate a
+     * new {@link Graph} using the {@link Supplier}, add it to the {@link Map} tracking {@link Graph}
+     * references, and return that {@link Graph}. 
+     */
+    public final Graph openGraph(final String graphName, final Supplier<Graph> supplier) {
+        final Graph graph = graphs.get(graphName);
+        if (null != graph) {
+            return graph;
+        }
+        final Graph newGraph = supplier.get();
+        addGraph(graphName, newGraph);
+        return newGraph;
+    }
+
+    /**
+     * Use {@link String} graphName to retrieve {@link Graph} from the {@link Map} tracking
+     * {@link Graph} references, close the {@link Graph}, and remove it from the {@link Map}.
+     */
+    public final void closeGraph(final String graphName) {
+        final Graph graph = getGraph(graphName);
+        graphs.remove(graphName);
+        closeGraph(graph);
+    }
+
+    /**
+     * Close {@link Graph} object.
+     */
+    public final void closeGraph(final Graph graph) {
+        try {
+            graph.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
If it will help make more sense to you, I can show you my WIP implementation for JanusGraph.  I hadn't brought it up to avoid clouding decisions made here based on the needs of a specific implementation.